hack/update-deps.sh

```diff
@@ -53,7 +53,7 @@ fi
 
 
 # Ensure we have everything we need under vendor/
-go mod tidy
+go mod tidy -compat=1.17
```

Comment:
@piyush-garg @pradeepitm12 please verify the correctness of this 🙃

Reply:
there is one more option `go mod tidy -go=1.16 && go mod tidy -go=1.17`; not sure what the right thing is

Reply:
btw, without this, `./hack/update-deps.sh` fails

Reply:
we need to keep `go mod tidy` normal, meaning we need to force the state to work with 1.17.. I'll try to take a look there.

Reply:
> there is one more option `go mod tidy -go=1.16 && go mod tidy -go=1.17`; not sure what the right thing is

I would go with this option yes

Reply:
What is the error.  If its not huge could you please paste it here.

Reply:
@pradeepitm12 `go mod tidy` prints this error:
```
To upgrade to the versions selected by go 1.16:
        go mod tidy -go=1.16 && go mod tidy -go=1.17
If reproducibility with go 1.16 is not needed:
        go mod tidy -compat=1.17
```

Reply:
I did what @vdemeester suggested and we don't need `-compat` anymore, I've included that in the commit https://github.com/tektoncd/cli/pull/1580/commits/e8d9831e5735af851be23d0d46d3170b19f3b686